trino-s3-proxy/src/test/java/io/trino/s3/proxy/server/TestProxiedRequestsWithVirtualHostProxy.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.s3.proxy.server;
+
+import com.google.inject.Inject;
+import io.trino.s3.proxy.server.testing.ManagedS3MockContainer.ForS3MockContainer;
+import io.trino.s3.proxy.server.testing.harness.TrinoS3ProxyTest;
+import io.trino.s3.proxy.server.testing.harness.TrinoS3ProxyTestCommonModules.WithConfiguredBuckets;
+import io.trino.s3.proxy.server.testing.harness.TrinoS3ProxyTestCommonModules.WithVirtualHostEnabledProxy;
+import software.amazon.awssdk.services.s3.S3Client;
+
+import java.util.List;
+
+@TrinoS3ProxyTest(modules = {WithConfiguredBuckets.class, WithVirtualHostEnabledProxy.class}, filters = WithVirtualHostEnabledProxy.class)
```

Comment:
nit: the testing builder has an `addModule` method so `WithVirtualHostEnabledProxy` as a module specified here isn't necessary. Maybe we don't even need modules in `TrinoS3ProxyTest` and can move everything to filters.